SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -739,9 +743,32 @@ int64 USpatialActorChannel::ReplicateActor()
 	if (NetDriver->LoadBalanceStrategy != nullptr &&
 		NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID))
 	{
-		if (!NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor) && !NetDriver->LockingPolicy->IsLocked(Actor))
+		if (MigrationVirtualWorkerId != SpatialConstants::INVALID_VIRTUAL_WORKER_ID)
+		{
+			if(MigrationFrame != NetDriver->ReplicationFrame)
+			{
+				Sender->SendAuthorityIntentUpdate(*Actor, MigrationVirtualWorkerId);
+
+				// If we're setting a different authority intent, preemptively changed to ROLE_SimulatedProxy
+				Actor->Role = ROLE_SimulatedProxy;
+				Actor->RemoteRole = ROLE_Authority;
+
+				Actor->OnAuthorityLost();
+				MigrationVirtualWorkerId = SpatialConstants::INVALID_VIRTUAL_WORKER_ID;
+			}
+		}
+		else if (!NetDriver->LoadBalanceStrategy->ShouldHaveAuthority(*Actor) && !NetDriver->LockingPolicy->IsLocked(Actor))
 		{
-			const AActor* NetOwner = Actor->GetNetOwner();
+			AActor* NetOwner = [this]
+			{
+				AActor* HierarchyRoot = SpatialGDK::GetHierarchyRoot(Actor);
```

Comment:
Does it affect the expectation of where GetHierarchyRoot is used elsewhere if the GetHierarchyRoot call returns the Actor itself if it has no owner? Would mean we can reduce these 5 lines to 1

Reply:
Mostly used in the LockingPolicy, and the code handles null cases in different ways.

Reply:
Did some refactor, renamed the current GetHierarchyRoot to GetTopmostOwner.